pkg/cluster/config/default.go

```diff
@@ -41,14 +41,30 @@ func SetDefaults_Cluster(obj *Cluster) {
 			},
 		}
 	}
-	// default to listening on 127.0.0.1:randomPort
-	// TODO(bentheelder): this defaulting will need to be ipv6 aware as well
-	if obj.Networking.APIServerAddress == "" {
+	if obj.Networking.IPFamily == "" {
+		obj.Networking.IPFamily = "ipv4"
+	}
+	// default to listening on 127.0.0.1:randomPort on ipv4
+	// docker has limitations to expose ports on IPv6 addresses
+	// TODO(bentheelder): restrict default for ipv6 clusters
+	if obj.Networking.APIServerAddress == "" && obj.Networking.IPFamily == "ipv4" {
 		obj.Networking.APIServerAddress = "127.0.0.1"
```

Comment:
we should default ipv6 to localhost as well?

Reply:
from a UX point of view, I suggest to keep it this way at least during some time.
It took me a lot of time today to realize that I can´t use localhost from the host to reach the endpoints, you have to use [::1] instead 😅 
This way you can reach the API using an IPv4 or an IPv6 socket this way, and the main goal now is to test the kubernetes IPv6 support, I think that we are good with the TODO

Reply:
if we don't set this it will bind to 0.0.0.0 which is insecure

Reply:
yeah, I always forget, I added your fix to the PR

Reply:
thanks :-)

Reply:
@BenTheElder this requires having localhost as an entry for `::1` , otherwise it fails
I hit this in the CircleCI and seems that's missing in old linux versions distributions.

```
# IP-Address  Full-Qualified-Hostname  Short-Hostname
#

127.0.0.1       localhost

# special IPv6 addresses
::1             localhost ipv6-localhost ipv6-loopback
```

Reply:
We just need to write the KUBECONFIG appropriately

Reply:
I have doubts, I was thinking the same at the beginning,
 but I think that localhost resolve to `::1`  too, I'm leaning keep it as is.

Reply:
in KUBECONFIG we can write the ip instead of localhost

Reply:
we do write the IP now